samples/net/openthread/ncp/overlay-tri-n4m-br.conf

```diff
@@ -0,0 +1,35 @@
+# TRI n4m BR
```

Comment:
Could you elaborate a bit what is this device? Is this some kind of shield? I've tried to google it, but quick search didn't provide anything useful.

I'm asking, because from user perspective I'd like to know **when** should this overlay be applied, what purpose does it serve. As far as I understand you run FRDM-K64F with 802.15.4 radio attached, but there's no information whatsoever in the documentation on what configuration is supported by this overlay.

Reply:
It is this device https://www.tridonic.com/com/en/products/net4more-borderROUTER-PoE-Thread.asp, also listed here: https://www.threadgroup.org/What-is-Thread#certifiedproducts.

The device internally uses a Linux CPU, NXP K64F (not the FRDM board), and AT86RF233.
It could be used to also use for the FRDM board with an ATREB233-XPRO with modifications. The overlay might serve as an example for this.

Did you need an overlay for nrf52840?

Reply:
> Did you need an overlay for nrf52840?

No, once I sorted out the compilation errors (as mentioned [here](https://github.com/zephyrproject-rtos/zephyr/pull/23249#discussion_r394352574)) the NCP worked in the default configuration (with docker image of BR).

I think it's worth to mention this configuration in the documentation then, see how it's done in `echo_client` for instance:
https://github.com/zephyrproject-rtos/zephyr/tree/master/samples/net/sockets/echo_client#building-and-running
Sadly not all overlays are documented in there, but it gives an overview of what's supported.

Some samples specify also a board-specific configs (https://github.com/zephyrproject-rtos/zephyr/tree/master/samples/net/sockets/echo_client/boards), but if it's a board+shield combo I'm not sure if it's any useful.

Reply:
I would add the following to README.rst.

```
There are configuration files for different boards and setups in the
echo-client directory:

- :file:`prj.conf`
  Generic config file, normally you should use this.

- :file:`overlay-tri-n4m-br.conf`
  This is an overlay for the dedicated Thread Border Router hardware 
  https://www.tridonic.com/com/en/products/net4more-borderROUTER-PoE-Thread.asp.
  The board support is not part of the Zephyr repositories, but the
  product is based on NXP K64 and AT86RF233. This file can be used as an
  example for a development set-up based on development boards.
```

Do you want to add a sentence on NRF?

Reply:
Just fix the first sentence, it's not `echo-client`.

I don't think you need to explicitly mention nRF here, I believe it's safe to assume that all OT/15.4 samples work on nRF (at least that's my goal).

Reply:
done and rebased.